src/blob_storage.cc

```diff
@@ -66,6 +99,29 @@ void BlobStorage::MarkFileObsolete(std::shared_ptr<BlobFileMeta> file,
   AddStats(stats_, cf_id_, TitanInternalStats::NUM_OBSOLETE_BLOB_FILE, 1);
 }
 
+bool BlobStorage::RemoveFile(uint64_t file_number) {
+  mutex_.AssertHeld();
+
+  auto file = files_.find(file_number);
+  if (file == files_.end()) {
+    return false;
+  }
+  // Removes from blob_ranges_
+  auto p = blob_ranges_.equal_range(file->second->smallest_key());
+  for (auto it = p.first; it != p.second; it++) {
+    if (it->second->file_number() == file->second->file_number()) {
+      it = blob_ranges_.erase(it);
+      break;
+    }
+  }
+  files_.erase(file_number);
+  file_cache_->Evict(file_number);
```

Comment:
Shall evict from cache before doing anything. Or does it matter?

Reply:
any difference? what's the matter?

Reply:
Sorry I didn't explain well. Can a reader accidentally obtain file handle from `file_cache_` after we remove the file from `file_` and if so will it cause error?

Reply:
`Evict` actually calls `cache_->Erase` which tells the underlying entry will be kept around until all existing handles to it have been released. So it would cause an error. 
And seems the case `a reader accidentally obtain file handle from file_cache_ after we remove the file from file_ ` would happen, cause we remove file only if all snapshot exceed the obsolete sequence.